File: Programming/ecobike_rental/lib/model/service/network/response/station.dart
import 'responses.dart';

class StationResponse {
  int stationId;
  int addressId;
  String stationName;
  String stationImage;
  String contactName;
  String email;
  String phone;
  int area;
  AddressResponse address;
  List<BikeResponse> listBike;

  StationResponse(
      {this.stationId,
      this.addressId,
      this.stationName,
      this.stationImage,
      this.contactName,
      this.email,
      this.phone,
      this.area,
      this.address,
      this.listBike});

  StationResponse.fromJson(Map<String, dynamic> json) {
    stationId = json['stationId'];
    addressId = json['addressId'];
    stationName = json['stationName'];
    stationImage = json['stationImage'];
    contactName = json['contactName'];
    email = json['email'];
    phone = json['phone'];
    area = json['area'];
    address = json['address'] != null
        ? AddressResponse.fromJson(json['address'])
        : null;
    if (json['listBike'] != null) {
      listBike = <BikeResponse>[];
      json['listBike'].forEach((v) {
        listBike.add(BikeResponse.fromJson(v));
      });
    }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['stationId'] = this.stationId;
    data['addressId'] = this.addressId;
    data['stationName'] = this.stationName;
    data['stationImage'] = this.stationImage;
    data['contactName'] = this.contactName;
    data['email'] = this.email;
    data['phone'] = this.phone;
    data['area'] = this.area;
    if (this.address != null) {
      data['address'] = this.address.toJson();
    }
    if (this.listBike != null) {
      data['listBike'] = this.listBike.map((v) => v.toJson()).toList();
    }
    return data;
  }
}

File: Programming/ecobike_rental_server/TKXDPM_API/Model/Station.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using AutoMapper;

namespace TKXDPM_API.Model
{
    public class Station
    {
        [Key] public int StationId { get; set; }

        [Column(TypeName = "varchar(255)")] public string StationName { get; set; }
        [Column(TypeName = "varchar(512)")] public string StationImage { get; set; }
        public int AddressId { get; set; }
        public Address Address { get; set; }
        [Column(TypeName = "varchar(255)")] public string ContactName { get; set; }
        [Column(TypeName = "varchar(255)")] public string Email { get; set; }
        [Column(TypeName = "varchar(255)")] public string Phone { get; set; }
        public float Area { get; set; }
        public List<BikeInStation> BikeInStations { get; set; }

        public IEnumerable<Bike> GetListBikes(BikeType type)
        {
            return (from bikeInStation in BikeInStations
                select bikeInStation.Bike
                into bike
                where bike.Type == type
                select bike);
        }
    }

    public class StationResponse
    {
        public static StationResponse CreateByMapper(Station station, IMapper mapper)
        {
            var stationResponse = mapper.Map<StationResponse>(station);
            if (station.Address != null)
            {
                stationResponse.Address = mapper.Map<AddressResponse>(station.Address);
            }

            var listBike = new List<BikeResponse>();
            foreach (var bikeInStation in station.BikeInStations)
            {
                listBike.Add(mapper.Map<BikeResponse>(bikeInStation.Bike));
            }

            stationResponse.ListBike = listBike;

            return stationResponse;
        }

        public int StationId { get; set; }
        public int AddressId { get; set; }
        public string StationName { get; set; }
        public string StationImage { get; set; }
        public string ContactName { get; set; }
        public string Email { get; set; }
        public string Phone { get; set; }
        public float Area { get; set; }
        public AddressResponse Address { get; set; }
        public List<BikeResponse> ListBike { get; set; }
    }
}